From 1ca8b5fc81f308b52d4b812a6094af86f2faedc3 Mon Sep 17 00:00:00 2001
From: Jorge Cisneros <jorge.cisneros@hpe.com>
Date: Tue, 3 Jun 2025 20:48:01 +0000
Subject: [PATCH] fix: increase buffer size in websocket connection
 implementation

Upstream-Status: Pending

Signed-off-by: Jorge Cisneros <jorge.cisneros@hpe.com>
---
 http/websocket_impl.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/http/websocket_impl.hpp b/http/websocket_impl.hpp
index 3cb5a994..2305e7b3 100644
--- a/http/websocket_impl.hpp
+++ b/http/websocket_impl.hpp
@@ -59,7 +59,7 @@ class ConnectionImpl : public Connection
             messageExHandlerIn,
         std::function<void(Connection&, const std::string&)> closeHandlerIn,
         std::function<void(Connection&)> errorHandlerIn) :
-        uri(urlViewIn), ws(std::move(adaptorIn)), inBuffer(inString, 131088),
+        uri(urlViewIn), ws(std::move(adaptorIn)), inBuffer(inString, 524352), // (128 * 1024 + 16) * 4 
         openHandler(std::move(openHandlerIn)),
         messageHandler(std::move(messageHandlerIn)),
         messageExHandler(std::move(messageExHandlerIn)),
