From 2389b888de7dca1647318c57ffde7db00d50c031 Mon Sep 17 00:00:00 2001
From: Ivan Farkas <ivan.farkas@hpe.com>
Date: Wed, 14 Feb 2024 09:44:18 -0800
Subject: [PATCH] Fixed exception when loading existing log entries

Upstream-Status: Pending
Signed-off-by: Ivan Farkas <ivan.farkas@hpe.com>
---
 log_manager.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/log_manager.cpp b/log_manager.cpp
index cf7ef60..89e4808 100644
--- a/log_manager.cpp
+++ b/log_manager.cpp
@@ -570,10 +570,10 @@ void Manager::restore()
 
     for (auto& file : fs::directory_iterator(dir))
     {
-        auto id = file.path().filename().c_str();
+        auto id = file.path().filename().string();
         auto idNum = std::stol(id);
         auto e = std::make_unique<Entry>(
-            busLog, std::string(OBJ_ENTRY) + '/' + id, idNum, *this);
+            busLog, std::string(OBJ_ENTRY) + "/" + id, idNum, *this);
         if (deserialize(file.path(), *e))
         {
             // validate the restored error entry id
